### PR TITLE
Introduce a CI Testing mode

### PR DESCRIPTION
**Pull request by AvdLee**
After we merged in this PR, we can add a `pr-testing` label to any PR which will allow to:
- Run CI as normal under draft
- Use the submodule CI reference instead of updating to latest `master` branch

This will improve our CI testing experience and makes it more clear for reviewers that a PR is work in progress. Before this, we had to open a PR as if it was ready to be reviewed, even though we might wanted to do some CI testing first.

### CODE REVIEW

**Comment by wetransferplatform** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  View more details on <a href="https://app.bitrise.io/build/cbc95730-4707-4700-90d8-fd2b4eae80db" target="_blank">Bitrise</a>
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>WeTransferPRLinterTests: Executed 19 tests (0 failed, 0 retried, 0 skipped) in 7.631 seconds</td>
    </tr>
  </tbody>
</table>

## Code Coverage Report
| Name | Coverage ||
| --- | --- | --- |
WeTransferPRLinter | 92.44% | ✅
WeTransferPRLinter | 0.00% | ⚠️

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against d6f7f489f0a09dd9112c5429407376a099fb96c4
</p>

**Review by BasThomas — APPROVED** [review]

File: Bitrise/Scripts/pr_assignments.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# It's first needed to reverse the URL, split it by `.` and get the second part of it (the -f2 in `cut`),
+# enabling to equate the URL split:
+#   - URL: https://github.com/WeTransfer/WeTransfer-iOS-CI.git
+#   - Reversed: tig.IC-SOi-refsnarTeW/refsnarTeW/moc.buhtig//:sptth
+#   - Sectioned: [1: tig].[2: IC-SOi-refsnarTeW/refsnarTeW/moc].[3: buhtig//:sptth]
+# The previous split is the input to the next one, this time by `/`, and then the first element is used (the -f1 in `cut`).
+#   - Input: IC-SOi-refsnarTeW/refsnarTeW/moc
+#   - Sectioned: [1: IC-SOi-refsnarTeW]/[2: refsnarTeW]/[3: moc]
+# And lastly, we revert the result one more time, as the result will be reversed otherwise.
+REPOSITORY_NAME=$(echo ${GIT_REPOSITORY_URL} | rev | cut -d '.' -f2 | cut -d '/' -f1 | rev)
```
Comment (peagasilva) [inline Bitrise/Scripts/pr_assignments.sh:15]: I wonder if making it a function to be reused would make sense here. What do you think?

File: Bitrise/testing_bitrise.yml [hunk 2]
```diff
@@ -9,63 +9,21 @@ workflows:
...
-            PR_AUTHOR=$(
-              curl \
-                -s \
-                -u ${GITBUDDY_ACCESS_TOKEN} \
-                -H ${HEADER} \
-                ${ISSUE_URL} \
-                | jq -r .user.login
-            )
+            # Assign a PR owner if not existing.
+            ./Bitrise/Scripts/pr_assignments.sh
```
Comment (peagasilva) [inline Bitrise/testing_bitrise.yml:21]: nit: should the directories be lower case the first letter (`./bitrise/scripts/pr_assignments.sh`)?

**Review by peagasilva — APPROVED** [review]

File: Bitrise/testing_bitrise.yml (hunk 2 above)
Comment (AvdLee) [reply Bitrise/testing_bitrise.yml:21]: Hmm, all folders currently have uppercase first. What would be a reason to change this?

File: Bitrise/Scripts/pr_assignments.sh (hunk 1 above)
Comment (AvdLee) [reply Bitrise/Scripts/pr_assignments.sh:15]: That makes sense, I'll see if I can make that work!

Comment (AvdLee) [reply Bitrise/Scripts/pr_assignments.sh]: [Added it here](https://github.com/WeTransfer/WeTransfer-iOS-CI/pull/207/commits/5f025e56a8a42a90852d81224746c7692b311205)

**Pull request merged**